hera_pspec/pspecbeam.py

```diff
@@ -0,0 +1,331 @@
+        but the results are frequency independent.
+
+        Currently, only the "pseudo Stokes I" beam is supported.
+        See Equations 4 and 5 of Moore et al. (2017) ApJ 836, 154
+        or arxiv:1502.05072 for details.
+
+        Parameters
+        ----------
+        stokes: str, optional
+                Which Stokes parameter's beam to compute the scalar for.
+                'pseudo_I', 'pseudo_Q', 'pseudo_U', 'pseudo_V', although currently only 'pseudo_I' is implemented
+                Default: 'pseudo_I'
+        Returns
+        -------
+        primary beam area: float, array-like
+        """
+        if stokes != 'pseudo_I':
+            raise NotImplementedError("Only stokes='pseudo_I' is currently supported.")
+        else:
+            return np.ones_like(self.beam_freqs) * np.pi * self.fwhm**2 / (8. * np.log(2.))
+
+    def compute_pspec_scalar(self, lower_freq, upper_freq, num_freqs, stokes='pseudo_I', taper='none', little_h=True, num_steps=10000):
+        """
+        Computes the scalar function to convert a power spectrum estimate
+        in "telescope units" to cosmological units
+
+        See arxiv:1304.4991 and HERA memo #27 for details.
+
+        Currently, only the "pseudo Stokes I" beam is supported.
+        See Equations 4 and 5 of Moore et al. (2017) ApJ 836, 154
+        or arxiv:1502.05072 for details.
+
+        Parameters
+        ----------
+        lower_freq: float
+                Bottom edge of frequency band over which power spectrum is
+                being estimated. Assumed to be in Hz.
+
+        upper_freq: float
+                Top edge of frequency band over which power spectrum is
+                being estimated. Assumed to be in Hz.
+
+        num_freqs: int
+                Number of frequency channels in the data.
+
+        stokes: str, optional
+                Which Stokes parameter's beam to compute the scalar for.
+                'pseudo_I', 'pseudo_Q', 'pseudo_U', 'pseudo_V', although currently only 'pseudo_I' is implemented
+                Default: 'pseudo_I'
+
+        taper : str, optional
+                Whether a tapering function (e.g. Blackman-Harris) is being
+                used in the power spectrum estimation.
+                Default: none
+
+        little_h : boolean, optional
+                Whether to have cosmological length units be h^-1 Mpc or Mpc
+                Value of h is obtained from conversion object stored in pspecbeam
+                Default: h^-1 Mpc
+
+        num_steps : int, optional
+                Number of steps to use when interpolating primary beams for
+                numerical integral
+                Default: 10000
+
+        Returns
+        -------
+        scalar: float
+                [\int dnu (\Omega_PP / \Omega_P^2) ( B_PP / B_P^2 ) / (X^2 Y)]^-1
+                in h^-3 Mpc^3 or Mpc^3.
+        """
+
+        integration_freqs = np.linspace(lower_freq,upper_freq,num_steps)
```

Comment:
scratch that, it should be `endpoint=False`

Reply:
Ok, I changed that. (It makes basically no difference, actually, because this does not correspond to the frequency channels of the measurement, but is just a grid in frequency to mimic a continuous integral).